lineoperations/src/lineoperations.c

```diff
@@ -160,27 +157,32 @@ action_indir_manip_item(GtkMenuItem *menuitem, gpointer gdata)
 	gint (*func)(gchar **lines, gint num_lines, gchar *new_file) = gdata;
 	GeanyDocument *doc    = document_get_current();
 
-	struct lo_lines sel   = get_current_sel_lines(doc->editor->sci);
-	gint   num_lines      = (sel.end_line - sel.start_line) + 1;
-
 	gint   num_chars      = 0;
 	gint   i              = 0;
 	gint   lines_affected = 0;
 
+	struct lo_lines *sel = g_malloc(sizeof(struct lo_lines));
```

Comment:
instead of allocating on the heap you could pass `&sel` to the `get_current_sel_lines()` call below